### DISCUSSION TITLE
Open *.gz (or other compressed files)

### DISCUSSION THREAD

**Discussion by prairie-guy** [nushell/nushell#12513] [General]
I am new to nushell. There is much I really like! 

I’m trying to customize it for my own use cases, one of which is working with large compressed files, most frequently compressed with `.gz` but also `tar.gz` Within nushell, it would be great to overload `open` so that I could have `.gz` files be recognized and opened natively. Here is what I do now:

```
gunzip -c ratio_converted_uniq_5mC164/B1.tsv.gz | from tsv
``` 
This works well, but I have other file types I would like to open with other files types as well:
```
def "open gz" [file] {gunzip -c $file} 
```

I’m sorry in advance if this question has been addressed in nushell documentation, within these discussions or online. I’ve looked . . .

**Comment by sholderbach** [comment]
Under the hood `open` will look for `from ...` commands which match the file ending. so `open file.tsv` is doing the file opening and then invokes `from tsv` with the default arguments. Since https://github.com/nushell/nushell/pull/10593 you can define a `from tsv.gz` that will decompress the stream and dispatch to `from tsv` in turn. 

```Nushell
def "from tsv.gz" [] {
  zcat | from tsv
}
```

**Comment by prairie-guy** [comment]
This is great. I defined my own “open gz”, but def “from tsv.gz” is better. Many thanks!

**Reply by sholderbach** [reply]
Defining `from gz` is still a useful idea, otherwise you have to define `from fastq.gz`, `from txt.gz`, ... but you can still specialize with a custom from like for `from bed.gz`

```
def "from gz" [] {
  zcat
}

def "from bed.gz" [] {
  zcat | from tsv --noheaders
}
```

```
open IMR-90_18_hg38.bed.gz | first 10
╭─#─┬column1┬column2─┬column3─┬───column4───╮
│ 0 │ chr1  │  10000 │  10600 │ 12_ZNF/Rpts │
│ 1 │ chr1  │  10600 │  13000 │ 6_TxWk      │
│ 2 │ chr1  │  13000 │  13200 │ 11_EnhWk    │
│ 3 │ chr1  │  13200 │  16000 │ 6_TxWk      │
│ 4 │ chr1  │  16000 │  17000 │ 5_Tx        │
│ 5 │ chr1  │  17000 │  19200 │ 6_TxWk      │
│ 6 │ chr1  │  19200 │  79000 │ 18_Quies    │
│ 7 │ chr1  │  79000 │  93600 │ 17_ReprPCWk │
│ 8 │ chr1  │ 605020 │ 605420 │ 1_TssA      │
│ 9 │ chr1  │ 605420 │ 725620 │ 18_Quies    │
╰───┴───────┴────────┴────────┴─────────────╯
```
